torchgeo/datasets/quakeset.py

```diff
@@ -0,0 +1,289 @@
+                        data.append(
+                            dict(
+                                key=k,
+                                patch=patch,
+                                images=("before", "pre"),
+                                label=0,
+                                magnitude=0.0,
+                            )
+                        )
+        f.close()
+        return data
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the image
+        """
+        import h5py
+
+        key = self.data[index]["key"]
+        patch = self.data[index]["patch"]
+        images = self.data[index]["images"]
+
+        with h5py.File(self.filepath) as f:
+            pre_array = f[key][patch][images[0]][:]
+            pre_array = np.nan_to_num(pre_array, nan=0)
+            post_array = f[key][patch][images[1]][:]
+            post_array = np.nan_to_num(post_array, nan=0)
+            array = np.concatenate([pre_array, post_array], axis=-1)
+            array = array.astype(np.float32)
+
+        tensor = torch.from_numpy(array)
+        # Convert from HxWxC to CxHxW
+        tensor = tensor.permute((2, 0, 1))
+        return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        # Check if the files already exist
+        if os.path.exists(self.filepath):
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        # Download the dataset
+        self._download()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        if not os.path.exists(self.filepath):
+            download_url(
+                self.url,
+                self.root,
+                filename=self.filename,
+                md5=self.md5 if self.checksum else None,
+            )
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: str | None = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional suptitle to use for figure
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        image = sample["image"].permute((1, 2, 0)).numpy()
+        label = cast(int, sample["label"].item())
+        label_class = self.classes[label]
+
+        # Create false color image for pre image
+        vv = percentile_normalization(image[..., 0]) + 1e-16
+        vh = percentile_normalization(image[..., 1]) + 1e-16
+        pre_fci = np.stack([vv, vh, vv / vh], axis=-1).clip(0, 1)
+
+        # Create false color image for post image
```

Comment:
What about before?

Reply:
Pre/post has different meanings here, it's just the images at t=1 and t=2. Maybe I can just rename as image1, image2?

Reply:
Changed to `fci1`, `fci2`